dts/arm/ti/am64x_r5.dtsi

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2025 Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv7-r.dtsi>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
+#include <zephyr/dt-bindings/interrupt-controller/ti-vim.h>
+#include "am64x_main.dtsi"
+#include "am64x_mcu.dtsi"
+
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-r5";
+			reg = <0>;
+			clock-frequency = <DT_FREQ_M(800)>;
+		};
+	};
+
+	atcm: memory@0 {
+		device_type = "memory";
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x0 DT_SIZE_K(32)>;
+		zephyr,memory-region = "ATCM";
+		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM) )>;
+	};
+
+	btcm: memory@41010000 {
+		device_type = "memory";
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x41010000 DT_SIZE_K(32)>;
+		zephyr,memory-region = "BTCM";
+		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM) )>;
+	};
+
+	sram: memory@70080000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x70080000 DT_SIZE_K(256)>;
+		zephyr,memory-region = "SRAM";
+		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM) )>;
+	};
+
+	vim: interrupt-controller@2fff0000 {
+		#address-cells = <1>;
+		compatible = "ti,vim";
+		reg = <0x2fff0000 0x4000>;
+		interrupt-controller;
+		#interrupt-cells = <4>; /* {IRQ/FIQ, IRQ_NUM, IRQ_TYPE, IRQ_PRIO} */
+		status = "okay";
+	};
+
+	/* Timer0 */
+	systick_timer: timer@2400000 {
+		compatible = "ti,am654-timer";
+		reg = <0x2400000 DT_SIZE_K(1)>;
+		interrupts = <0 152 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-parent = <&vim>;
+		status = "okay";
+	};
+
+	/* users: r5f0_0, r5f0_1, r5f1_0, r5f1_1 */
+	mbox0: mailbox@29000000 {
```

Comment:
All these mailbox nodes could go into `am64x_main.dtsi`, the only thing that changes based on core is `interrupts/interrupt-parent` which can stay here. Same for M4, it uses the same `mbox6` as defined here, but with different interrupt integration, it can fetch the base definition from the dtsi file.

Reply:
I put mbox6 and mbox7 in `am64x_mcu.dtsi` since that way those nodes are included by both `am64x_r5.dtsi`  and `am64x_m4.dtsi` (and potentially a53). Rest of the mailboxes are in `am64x_main.dtsi` which is not included by m4.

Reply:
`mbox6` and `mbox7` are not in the MCU domain, so why have them in `am64x_mcu.dtsi`? If this is all to keep `am64x_m4.dtsi` from include `am64x_main.dts` I'm not sure what that is getting you.

Reply:
Yes, you can include the MAIN domain dtsi file from M4, and if folks want to use the M4 in isolation from any MAIN domain peripheral, all they have to do is not use them. Including the dtsi doesn't force the use of anything inside it, so that is fine.